sodym/named_dim_array_helper.py

```diff
@@ -0,0 +1,35 @@
+import numpy as np
+
+from .named_dim_arrays import Flow, NamedDimArray
+from .dimensions import Dimension, DimensionSet
+
+
+def named_dim_array_stack(named_dim_arrays: list[NamedDimArray], dimension: Dimension) -> NamedDimArray:
+    """Stack a list of NamedDimArray objects using a new dimension.
+    Like numpy.stack with axis=-1, but for `NamedDimArray`s.
+    Method can be applied to `NamedDimArray`s, `StockArray`s, `Parameter`s and `Flow`s.
+    """
+    named_dim_array0 = named_dim_arrays[0]
+    if len(named_dim_arrays) != dimension.len:
+        raise ValueError(
+            'Number of objects to stack must be equal to length of new dimension'
+        )
+    for named_dim_array in named_dim_arrays[1:]:
+        if named_dim_array.dims != named_dim_array0.dims:
+            raise ValueError(
+                'Existing dimensions must be identical in all objects to be stacked.'
+            )
+    axis = len(named_dim_array0.dims.dimensions)
+    extended_values = np.stack(
+        [named_dim_array.values for named_dim_array in named_dim_arrays], axis=axis
+    )
+    extended_dimensions = DimensionSet(dimensions=named_dim_array0.dims.dimensions+[dimension])
```

Comment:
with #32, you can use `named_dim_array0.dims.expand_by([dimension])`.

Reply:
I would go with merging this pr first, and then adding your suggested change to #32, if that's okay?